packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -211,6 +211,15 @@ class HotRunner extends ResidentRunner {
       }
     }
 
+    if (debuggingOptions.fastStart) {
```

Comment:
Consider a comment here, like "In fastStart mode, apps are initialized from a placeholder splashscreen app. We must do a restart here to load the program and assets for the real app".

Reply:
Done